core/trino-parser/src/main/java/io/trino/sql/tree/ArithmeticBinaryExpression.java

```diff
@@ -110,7 +111,10 @@ public boolean equals(Object o)
     @Override
     public int hashCode()
     {
-        return Objects.hash(operator, left, right);
+        if (hash == 0) {
+            hash = Objects.hash(operator, left, right);
+        }
```

Comment:
nit: https://github.com/trinodb/trino/blob/90c22f66c25e6e554b167773a2f706a9f137e4ab/core/trino-spi/src/main/java/io/trino/spi/type/TypeSignature.java#L171-L173

Reply:
I don't know, the probability of getting hash 0 is small... For example, Java 11 doesn't care about this for String, while Java 16 does...